streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -288,6 +291,8 @@ public StreamThread(TopologyBuilder builder,
         this.standbyRecords = new HashMap<>();
 
         this.stateDirectory = new StateDirectory(applicationId, config.getString(StreamsConfig.STATE_DIR_CONFIG), time);
+        Object maxPollInterval = consumerConfigs.get(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG);
+        this.rebalanceTimeoutMs =  maxPollInterval == null ? 300000 : (Integer) ConfigDef.parseType(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, maxPollInterval, Type.INT);
```

Comment:
Well I checked that code but please check this line
https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java#L550
Here `configNames` are consumer config names, however `originals()` is still defaulted by  `StreamsConfig#CONFIG ` as per line https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java#L217 and https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java#L449.
Now this `StreamsConfig#CONFIG ` does not have `MAX_POLL_INTERVAL_MS_CONFIG` defined by default, so if user does not provide this property in streams config, then it will never find in the `StreamsConfig#CONFIG ` map. Hence I had to check against the default value in `ConsumerConfig` if it is not provided.